source/common/http/http3/quic_client_connection_factory.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/config/listener/v3/quic_config.pb.h"
+#include "envoy/config/typed_config.h"
+#include "envoy/network/connection.h"
+#include "envoy/ssl/context_config.h"
+
+namespace Envoy {
+namespace Http {
+
+// A factory to create EnvoyQuicClientConnection instance for QUIC
```

Comment:
nits: s/EnvoyQuicClientConnection/EnvoyQuicClientSession and  EnvoyQuicClientConnection